conda/core/index.py

```diff
@@ -35,17 +35,37 @@ def check_allowlist(channel_urls: list[str]) -> None:
 
     :param channel_urls: A list of channel URLs to check against the allowlist.
     :raises ChannelNotAllowed: If any URL is not in the allowlist.
+    :raises ChannelDenied: If any URL is in the denylist.
     """
     if context.allowlist_channels:
         allowlist_channel_urls = tuple(
             chain.from_iterable(
-                Channel(c).base_urls for c in context.allowlist_channels
+                Channel(allowlist_channel).base_urls
+                for allowlist_channel in context.allowlist_channels
             )
         )
-        for url in channel_urls:
-            these_urls = Channel(url).base_urls
-            if not all(this_url in allowlist_channel_urls for this_url in these_urls):
-                raise ChannelNotAllowed(Channel(url))
+        for channel_url in channel_urls:
+            channel_base_urls = Channel(channel_url).base_urls
+            if not all(
+                channel_base_url in allowlist_channel_urls
+                for channel_base_url in channel_base_urls
+            ):
+                raise ChannelNotAllowed(Channel(channel_url))
+
+    elif context.denylist_channels:
```

Comment:
If they are mutually exclusive, I think we should error out early so `denylist` is not ignored, and this should be documented in their `context` docstrings.

Reply:
Nah, this is an oversight only, they should not be mutually exclusive